service-base/src/main/java/org/eclipse/hono/service/tenant/TenantAmqpEndpoint.java

```diff
@@ -53,28 +55,76 @@ public final String getName() {
+     * Checks if the client is authorized to invoke an operation.
      * <p>
-     * This check makes use of the provided tenantId by enriching the resource with it.
-     * So permission checks can be done on a per tenant level, although the endpoint the client
-     * connects to does not include the tenantId (like for several other of Hono's APIs).
+     * If the request does not include a <em>tenant_id</em> application property
+     * then the request is authorized by default. This behavior allows clients to
+     * invoke operations that do not require a tenant ID as a parameter. In such
+     * cases the {@link #filterResponse(HonoUser, EventBusMessage)} method is used
+     * to verify that the response only contains data that the client is authorized
+     * to retrieve.
+     * <p>
+     * If the request does contain a tenant ID parameter in its application properties
+     * then this tenant ID is used for the authorization check together with the
+     * endpoint and operation name.
      *
      * @param clientPrincipal The client.
      * @param resource The resource the operation belongs to.
-     * @param message The message for which the authorization shall be checked.
+     * @param request The message for which the authorization shall be checked.
      * @return The outcome of the check.
      * @throws NullPointerException if any of the parameters is {@code null}.
      */
     @Override
-    protected Future<Boolean> isAuthorized(final HonoUser clientPrincipal, final ResourceIdentifier resource, final Message message) {
+    protected Future<Boolean> isAuthorized(final HonoUser clientPrincipal, final ResourceIdentifier resource, final Message request) {
 
-        Objects.requireNonNull(message);
+        Objects.requireNonNull(request);
 
-        final String tenantId = MessageHelper.getTenantId(message);
-        final ResourceIdentifier specificTenantAddress =
-                ResourceIdentifier.from(resource.getEndpoint(), tenantId, null);
+        final String tenantId = MessageHelper.getTenantId(request);
+        if (tenantId == null) {
+            // delegate authorization check to filterResource operation
+            return Future.succeededFuture(Boolean.TRUE);
+        } else {
+            final ResourceIdentifier specificTenantAddress =
+                    ResourceIdentifier.fromPath(new String[] { resource.getEndpoint(), tenantId });
 
-        return getAuthorizationService().isAuthorized(clientPrincipal, specificTenantAddress, message.getSubject());
+            return getAuthorizationService().isAuthorized(clientPrincipal, specificTenantAddress, request.getSubject());
+        }
+    }
+
+    /**
+     * Verifies that a response only contains tenant information that the
+     * client is authorized to retrieve.
+     * <p>
+     * If the response does not contain a tenant ID nor a payload, then the
+     * response is returned as-is.
```

Comment:
I would prefer `unchanged` over `as-is`.

Reply:
I think that *as-is* is the right term here because we never change the payload. We fail the future if the client is not authorize to retrieve the data, but otherwise we return the payload *as-is*. Using *unchanged* instead could suggest that the payload's content is *filtered*, i.e. we would remove some of the information from the payload, which is not the case. So IMHO it should stay like it is now.